src/NServiceBus.RavenDB/Internal/DocumentStoreInitializer.cs

```diff
@@ -53,40 +53,66 @@ void ApplyConventions(ReadOnlySettings settings)
             }
-            if (!(store.TransactionRecoveryStorage is VolatileOnlyTransactionRecoveryStorage))
+            else
             {
-                Logger.Warn("Overriding user-specified documentStore.TransactionRecoveryStorage. It's no longer necessary to set this value while using NServiceBus.RavenDB persistence. Consider using busConfiguration.CustomizeDocumentStore(Action<IDocumentStore customize) if this is absolutely necessary, but it is not recommended.");
-            }
-
-            var resourceManagerId = settings.Get<string>("NServiceBus.LocalAddress") + "-" + settings.Get<string>("EndpointVersion");
-            store.ResourceManagerId = DeterministicGuidBuilder(resourceManagerId);
+                // Source: https://github.com/ravendb/ravendb/blob/f56963f23f54b5535eba4f043fb84d5145b11b1d/Raven.Client.Lightweight/Document/DocumentStore.cs#L129
+                var ravenDefaultResourceManagerId = new Guid("e749baa6-6f76-4eef-a069-40a4378954f8");
 
-            string path = $@"%LOCALAPPDATA%\NServiceBus.RavenDB\{store.ResourceManagerId}";
-            store.TransactionRecoveryStorage = new LocalDirectoryTransactionRecoveryStorage(path);
 
-            bool suppressDistributedTransactions;
-            if (settings.TryGet("Transactions.SuppressDistributedTransactions", out suppressDistributedTransactions) && suppressDistributedTransactions)
-            {
-                store.EnlistInDistributedTransactions = false;
+                if (store.JsonRequestFactory != null) // code for "has the DocStore been Initialized yet"
+                {
+                    // If the DocumentStore has already been initialized, that means transaction recovery has already
+                    // been run and the TransactionRecoveryStorage cannot be safely swapped with our safe values.
+                    // Only (known) reason to do this would be if customer wants to share the DocumentStore with
+                    // a wider application's persistence needs and must initialize it before NServiceBus is invoked.
+                    // Therefore, we will check to make sure ResourceManagerId and TransactionRecoveryStorage are
+                    // acceptable and throw if they are not.
+
+                    var dtcSettingsAcceptable = store.ResourceManagerId != Guid.Empty
+                                                && store.ResourceManagerId != ravenDefaultResourceManagerId
+                                                && store.TransactionRecoveryStorage is LocalDirectoryTransactionRecoveryStorage;
+
+                    if (!dtcSettingsAcceptable)
+                    {
+                        throw new InvalidOperationException("RavenDB Persistence DocumentStore has already been initialized without safe DTC settings which can lead to data loss. Either allow NServiceBus to initialize the DocumentStore instance or refer to documentation to set up safe DTC settings.");
```

Comment:
so we are basically saying: since we are going to use your store to persist also NServiceBus related data we want to be sure that we are safe enough so that we cannot run in any trouble due to mis-configured RavenDB TX-related settings.
:+1: